be/src/exprs/dict_query_expr.cpp

```diff
@@ -45,6 +45,9 @@ StatusOr<ColumnPtr> DictQueryExpr::evaluate_checked(ExprContext* context, Chunk*
     }
     ChunkPtr key_chunk = ChunkHelper::new_chunk(_key_schema, size);
     key_chunk->reset();
+    // make sure all columns in key_chunk have the same size.
+    // Otherwise assert fail in DEBUG/ASAN mode in check_or_die()
+    key_chunk->set_num_rows(size);
```

Comment:
After key_chunk->reset(), all the size of columns is 0, why check failed?

Reply:
In the following, we will call update_column_by_index() to replace the column one by one. Support in the init state, all column is zero size. But after we replace the first column, column_1.size > 0 which is not 0 but it will still call check_or_die() and cause (column_1.size != column_2.size(which is 0)) and fail

Reply:
Remove the check_or_die()? otherwise the set_num_rows will resize the column, later the column will release, the resize is useless?

Reply:
updated